plugins/modules/digital_ocean_domain_record_info.py

```diff
@@ -0,0 +1,227 @@
+        self.module = module
+        self.domain = module.params.get("name").lower()
+        self.records = self.__get_all_records()
+        self.payload = self.__build_payload()
+        self.force_update = module.params.get("force_update", False)
+        self.record_id = module.params.get("record_id", None)
+        self.records_by_id = self.__find_record_by_id(self.record_id)
+
+    def check_credentials(self):
+        # Check if oauth_token is valid or not
+        response = self.get("account")
+        if response.status_code == 401:
+            self.module.fail_json(
+                msg="Failed to login using oauth_token, please verify validity of oauth_token"
+            )
+
+    def __get_all_records(self):
+
+        records = []
+        page = 1
+        while True:
+            # GET /v2/domains/$DOMAIN_NAME/records
+            type = self.module.params.get("type")
+            if type:
+                response = self.get(
+                    "domains/%(domain)s/records?type=%(type)s&page=%(page)s"
+                    % {"domain": self.domain, "type": type, "page": page}
+                )
+            else:
+                response = self.get(
+                    "domains/%(domain)s/records?page=%(page)s"
+                    % {"domain": self.domain, "page": page}
+                )
+            status_code = response.status_code
+            json = response.json
+
+            if status_code != 200:
+                self.module.exit_json(
+                    msg="Error getting domain records [%(status_code)s: %(json)s]"
+                    % {"status_code": status_code, "json": json}
+                )
+
+            for record in json["domain_records"]:
+                records.append(dict([(str(k), v) for k, v in record.items()]))
+
+            if "pages" in json["links"] and "next" in json["links"]["pages"]:
+                page += 1
+            else:
+                break
+
+        return records
+
+    def get_records(self):
+        return False, self.records
+
+    def get_records_by_id(self):
+        if self.records_by_id:
+            return False, [self.records_by_id]
+        else:
+            return False, []
+
+    def __find_record_by_id(self, record_id):
+        for record in self.records:
+            if record["id"] == record_id:
+                return record
+        return None
+
+    def __build_payload(self):
+
+        payload = dict(
+            data=self.module.params.get("data"),
+            flags=self.module.params.get("flags"),
+            name=self.module.params.get("name"),
+            port=self.module.params.get("port"),
+            priority=self.module.params.get("priority"),
+            type=self.module.params.get("type"),
+            tag=self.module.params.get("tag"),
+            ttl=self.module.params.get("ttl"),
+            weight=self.module.params.get("weight"),
+        )
+
+        # DigitalOcean stores every data in lowercase except TXT
```

Comment:
I don't really understand why we need to `.lower()` all the things here. I understand that `TXT` might seem like a special case, but I there are [a lot of other record types](https://en.wikipedia.org/wiki/List_of_DNS_record_types) and I feel like the general case should be "pass the user's input through unchanged".

Do you think it will cause a problem to _not_ call `.lower()`?

Reply:
The majority of this code was copied from the existing `digital_ocean_domain` module, including this section. With that in mind, I'd rather leave it alone (versus having a difference here and not there, so to speak). Though, if the comment ("DigitalOcean stores every data in lowercase except TXT") is true, I think that's actually okay since the [DNS specification](https://datatracker.ietf.org/doc/html/rfc4343) is "case insensitive".

Reply:
The DNS specification is case insensitive in that the _name_ field is case insensitive (lookups for `digitalocean.com`, `DIGITALOCEAN.COM`, and `DiGiTAlOcEaN.CoM` will return identical results), but case might be important in non-`TXT` record types, and I'm hesitant to change user intent. For example, a `DNSKEY` resource record seems to very much be case sensitive, according to [RFC 4034](https://www.rfc-editor.org/rfc/rfc4034.html#section-2.3).

Reply:
This may be a moot point, for now, because I don't know if there are non-`TXT` DNS records that the DO API supports that are, in fact, case sensitive. But since this is a new module, I feel like it might be better to leave this out?
On the other hand, why are we even sending in `data` here? This is an informational module, correct? (So it shouldn't be updating the `data` field, or mangling what comes back from the DO API?)

Reply:
I've pushed changes which remove the unused params, and, the `lower()`.